client/maps/action-creators.ts

```diff
@@ -0,0 +1,248 @@
+import {
+  GetMapDetailsPayload,
+  GetMapsPayload,
+  MapInfoJson,
+  MapPreferences,
+  MapSortType,
+  MapVisibility,
+  Tileset,
+  UpdateMapPayload,
+  UpdateMapServerBody,
+  UploadMapPayload,
+} from '../../common/maps'
+import { ThunkAction } from '../dispatch-registry'
+import fetch from '../network/fetch'
+import { apiUrl } from '../network/urls'
+import { openSnackbar } from '../snackbars/action-creators'
+import { ClearMaps } from './actions'
+
+const uploadModule = IS_ELECTRON ? import('./upload') : Promise.resolve()
+
+export function uploadLocalMap(path: string, onMapSelect: (map: MapInfoJson) => void): ThunkAction {
+  return async dispatch => {
```

Comment:
Sorry I didn't notice this before, don't make thunks that return Promises, this isn't safe. Nothing will be attached to handle errors on this. You seemed to have realized this since you're no longer awaiting anything, but the async keyword still needs to be removed